### PR TITLE
Expand ledger api

### PR DESCRIPTION
**Pull request by lehins**
# Description

Add your description here, if it fixes a particular issue please provide a
[link](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword=)
to the issue.

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated
- [x] Any changes are noted in the `CHANGELOG.md` for affected package
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) (use `scripts/fourmolize.sh`)
- [x] Cabal files are formatted (use `scripts/cabal-format.sh`)
- [x] [`hie.yaml`](https://github.com/input-output-hk/cardano-ledger/blob/master/hie.yaml) has been updated (use `scripts/gen-hie.sh`)
- [x] Self-reviewed the diff

### CODE REVIEW

File: eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Scripts.hs [hunk 1]
```diff
@@ -146,7 +146,7 @@ data Tag
     Mint
   | -- | Validates certificate transactions
     Cert
-  | -- | Validates withdrawl from a reward account
+  | -- | Validates withdrawal from a reward account
```
Comment (JaredCorduan) [inline eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Scripts.hs:149]: I have a feeling this spelling mistake could be all over the place :grimacing:

File: eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxWits.hs [hunk 2]
```diff
@@ -231,8 +220,8 @@ instance (Era era, Script era ~ AlonzoScript era) => Semigroup (AlonzoTxWits era
   (<>) x y | isEmptyTxWitness x = y
   (<>) x y | isEmptyTxWitness y = x
   (<>)
-    (getMemoRawType -> AlonzoTxWitsRaw a b c d (Redeemers' e))
-    (getMemoRawType -> AlonzoTxWitsRaw u v w x (Redeemers' y)) =
+    (getMemoRawType -> AlonzoTxWitsRaw a b c d (Redeemers e))
```
Comment (JaredCorduan) [inline eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxWits.hs]: the master delouser at work!

File: eras/conway/impl/src/Cardano/Ledger/Conway/Governance.hs [hunk 3]
```diff
@@ -161,6 +166,91 @@ govActionIdToText (GovernanceActionId (TxId txidHash) (GovernanceActionIx ix)) =
...
+data Anchor c = Anchor
+  { anchorUrl :: !Url
+  , anchorDataHash :: !(SafeHash c AnchorDataHash)
+  }
+  deriving (Eq, Show, Generic)
+
+instance NoThunks (Anchor c)
+
+instance Crypto c => NFData (Anchor c) where
+  rnf = rwhnf
```
Comment (JaredCorduan) [inline eras/conway/impl/src/Cardano/Ledger/Conway/Governance.hs]: I'm curious why you didn't auto-derive this.

File: libs/cardano-ledger-api/src/Cardano/Ledger/Api/Governance.hs [hunk 4]
```diff
@@ -0,0 +1,84 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+
+module Cardano.Ledger.Api.Governance (
```
Comment (JaredCorduan) [inline libs/cardano-ledger-api/src/Cardano/Ledger/Api/Governance.hs]: :tada:

File: libs/cardano-ledger-test/cardano-ledger-test.cabal [hunk 5]
```diff
@@ -61,21 +61,22 @@ library
...
+        cardano-ledger-binary:{cardano-ledger-binary, testlib},
+        cardano-ledger-conway:{cardano-ledger-conway, testlib},
+        cardano-ledger-conway-test,
+        cardano-ledger-core:{cardano-ledger-core, testlib},
+        cardano-ledger-pretty,
+        cardano-ledger-mary,
+        cardano-ledger-shelley,
+        cardano-ledger-shelley-test,
+        cardano-ledger-shelley-ma-test,
+        cardano-protocol-tpraos,
```
Comment (JaredCorduan) [inline libs/cardano-ledger-test/cardano-ledger-test.cabal:79]: why did you remove the bounds?

**Review by JaredCorduan — APPROVED** [review]
exciting stuff!!

File: eras/conway/impl/src/Cardano/Ledger/Conway/Governance.hs (hunk 3 above)
Comment (lehins) [reply eras/conway/impl/src/Cardano/Ledger/Conway/Governance.hs]: Few reasons.

1. It is easy to do. Just one line.
2. It is then clear that the type is alwasy in NF
3. There is no runtime overhead due to generic deriving. NFData is mostly (maybe even only) used in benchmarking, so having no overhead is beneficial, but not required. 

Because of all those reasons I always prefer to defined NFData manually. I just don't always have time to do it.

File: libs/cardano-ledger-test/cardano-ledger-test.cabal (hunk 5 above)
Comment (lehins) [reply libs/cardano-ledger-test/cardano-ledger-test.cabal:79]: `cardano-ledger-test` is never released, so it will always use the packages that are in the repo, which is defined by `cabal.project`

In other words, bounds are only useful for packages that are being drawn from CHaPs or hackage. So, in order to avoid redundant work it is best to remove them

File: eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Scripts.hs (hunk 1 above)
Comment (lehins) [reply eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Scripts.hs:149]: :rofl: You are right:
```shell
cardano-ledger> git grep withdrawl | wc -l
5
cardano-node> git grep withdrawl | wc -l
2
```

Comment (lehins) [reply eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Scripts.hs:149]: Fixed it in all places

File: eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxWits.hs (hunk 2 above)
Comment (lehins) [reply eras/alonzo/impl/src/Cardano/Ledger/Alonzo/TxWits.hs]: Thank you! Today I learned a new word :smile:

File: eras/conway/impl/src/Cardano/Ledger/Conway/Governance.hs (hunk 3 above)
Comment (JaredCorduan) [reply eras/conway/impl/src/Cardano/Ledger/Conway/Governance.hs]: thanks for the explanation!
